src/main/scala/rise/core/makeClosed.scala

```diff
@@ -6,36 +6,24 @@ import rise.core.types._
 object makeClosed {
   def apply(e: Expr): Expr = withCount(e)._1
   def withCount(e: Expr): (Expr, Int) = {
-    val emptySubs: (Map[Type, Type],
-      Map[NatIdentifier, Nat],
-      Map[AddressSpaceIdentifier, AddressSpace],
-      Map[NatToDataIdentifier, NatToData]) = (Map(), Map(), Map(), Map())
-
-    val (expr, (ts, ns, as, n2ds)) = IsClosedForm.varsToClose(e)._2.foldLeft((e, emptySubs))((acc, ftv) => acc match {
-      case (expr, (ts, ns, as, n2ds)) => ftv match {
+    val (_, ftvs) = IsClosedForm.varsToClose(e)
+    val (expr, ts) = ftvs.foldLeft((e, Map[Type, Type]()))((acc, ftv) => acc match {
+      case (expr, ts) => ftv match {
         case i: TypeIdentifier =>
-          val dt = DataTypeIdentifier(freshName("dt"), isExplicit = true)
-          (DepLambda[DataKind](dt, expr)(DepFunType[DataKind, Type](dt, expr.t)),
-            (ts ++ Map(i -> dt), ns, as , n2ds))
+          val dt = DataTypeIdentifier(freshName("dt"))
+          (DepLambda[DataKind](dt, expr)(DepFunType[DataKind, Type](dt, expr.t)), (ts ++ Map(i -> dt)))
```

Comment:
Do we have tests that rely on this code path? We might want to allow dependent lambdas binding types in the future, changing to a data type identifiers seems like a hack we should get rid of.

Reply:
That seems to be the case. Initially things blew up because some traversal paths for dependent lambdas on type identifiers were missing, once I added those the trouble seemed to be with the fission tests in the fissionFussion suite and the simple fusion test in the traversals suite: they use `mapLastFission`, which only works on data types, weakening this condition breaks inference on the rewritten term. I will open a ticket.